py/server/deephaven/__init__.py

```diff
@@ -7,7 +7,9 @@
 
 """
 
-__version__ = "0.17.0"
+import os
+
+__version__ = os.environ.get('DEEPHAVEN_VERSION') or "0.17.0"
```

Comment:
doesn't this run the risk of installing version X, but having DEEPHAVEN_VERSION=Y set, and so the library reports the wrong version when asked?

Reply:
That's actually kinda the point. When the deploy tool pushes a deployment, it should report the version number used to deploy the system.  For the main demo controller, this lets us bump the versions w/o touching main versions. For single machine deployments used to test, we also stick `_branch-name` into the version as a suffix for we can tell where each machine came from, and so `print(deephaven.__version__)` will give the more useful value.

Reply:
I set it to NoDocumentationNeeded... we want it to be possible, but not really advertised. imho.

Reply:
It seems like we should move the logic to the `print()` statement if that's the purpose:

```
print(os.environ.get('DEEPHAVEN_VERSION') or deephaven.__version__)
```

Reply:
I am with Colin,  would prefer not to have the built-in package version number contingent on an env var.

Reply:
I can't tell the user running code in console to check the env var. That defeats the purpose.

The idea is that if I build some stuff on top of a deephaven base version, I want to tell user they have built a custom version and not the packaged version the code was based on.

Reply:
When I deploy a machine / cluster, I _DO_ set a custom version for all the other artifacts and those versions are respected, but I can't touch this one b/c it's hardcoded.  Unless we want to template / find replace the hardcoded version when building the code, I have no other way to indicate "this is not the version you actually built".

w/o this admittedly hacky method of injecting a version number, then I _can't_ update the version the console reports without manually editing this file.  Everything else I can route through a gradle property, but there's nothing else I can think of here (b/c editing the file before building wheel is super-duper hacky).

Reply:
So, the goal is to be able to detect both a) the version of the `deephaven-core` wheel and b) the version of the demo installation? As there is no actual `deephaven-core` wheel with version x.y.z_BRANCH_NAME, wouldn't it make more sense to actually want them to ask specifically for the branch name, rather than have the version reflect a version that doesn't exist?

If this isn't something we have scripted, but something we ask the user to do, can't we introduce a non-standard property/def instead? something like 
```
deephaven.printVersion()
```
which could then wrap up the custom logic you're after, and potentially also log details like the jvm version, etc?

Reply:
I added `deephaven.print_version()` since I assume that's "more pythonic"